### PR TITLE
feat: dial, drop and retrieve connected peers

### PR DESCRIPTION
**Pull request by gabrielmer**
Integrating and implementing features to dial, drop and retrieve connected peers. 

Important changes:
- [x] implemented DialPeer(), DropPeer() and `PeerCount()` functions
- [x] renamed GetNumConnectedPeers to `GetNumConnectedRelayPeers` 
- [x] added test for the dialing feature
- [x] fixed flaky test 

This PR depends on the new libwaku procedures exposed in https://github.com/waku-org/nwaku/pull/3149
Once the nwaku PR is merged, I will update the nwaku submodule in this PR

Issue: https://github.com/waku-org/nwaku/issues/3076

### CODE REVIEW

File: wakuv2/nwaku.go [hunk 1]
```diff
@@ -1895,10 +1912,9 @@ func (w *Waku) ClearEnvelopesCache() {
 	w.envelopeCache = newTTLCache()
 }
 
-// TODO-nwaku
 func (w *Waku) PeerCount() int {
-	return 0
-	// return w.node.PeerCount()
+	peerCount, _ := w.GetNumConnectedPeers()
```
Comment (richard-ramos) [inline wakuv2/nwaku.go]: I think in this case we shouldn't swallow the error, but instead modify the Signature of `func(w *Waku) PeerCount()` to return `(int, err)` instead

File: wakuv2/nwaku.go [hunk 2]
```diff
@@ -2165,22 +2181,27 @@ func (w *Waku) AddRelayPeer(address multiaddr.Multiaddr) (peer.ID, error) {
 }
 
 func (w *Waku) DialPeer(address multiaddr.Multiaddr) error {
-	// TODO-nwaku
-	/*
-		ctx, cancel := context.WithTimeout(w.ctx, requestTimeout)
-		defer cancel()
-		return w.node.DialPeerWithMultiAddress(ctx, address) */
-	return nil
+	return w.WakuDialPeer(address, "", 1000)
```
Comment (richard-ramos) [inline wakuv2/nwaku.go]: Instead of hardcoding the 1000, we can use instead `requestTimeout / time.Millisecond`

**Review by richard-ramos — APPROVED** [review]
LGTM! just left some minor comments

File: wakuv2/nwaku.go (hunk 1 above)
Comment (gabrielmer) [reply wakuv2/nwaku.go]: Sounds great! Will update now :))

Thank you!

File: wakuv2/nwaku.go (hunk 2 above)
Comment (gabrielmer) [reply wakuv2/nwaku.go]: ooh that looks much better!

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100:

**Comment by gabrielmer** [comment]
Closing this PR in favor of https://github.com/status-im/status-go/pull/6013 , as there are big conflicts with the base branch

**Pull request closed without merging**